crates/gitbutler-repo/src/repository_ext.rs

```diff
@@ -241,16 +254,19 @@ impl RepositoryExt for git2::Repository {
                     let blob = self.blob(path_str.as_bytes())?;
                     tree_update_builder.upsert(path, blob, git2::FileMode::Link);
                 } else {
-                    let file = std::fs::read(&file_path)?;
-                    let blob = self.blob(&file)?;
+                    let mut file_for_git =
+                        pipeline.convert_to_git(std::fs::File::open(&file_path)?, path, &index)?;
+                    buf.clear();
+                    std::io::copy(&mut file_for_git, &mut buf)?;
+                    let blob_id = self.blob(&buf)?;
```

Comment:
Why do we want to copy the data into the buffer and not just directly pass it to `self.blob`?

Reply:
`file_for_git` isn't a buffer, it's something that implements `std::io::Read`, while `blob()` takes `&[u8]`.

This is the version of the code that is easiest to write, but also slowest as two out of three cases already have a buffer underneath which could be passed directly.

I will adjust it to be the most efficient possible.